python/nav/web/api/v1/views.py

```diff
@@ -1153,3 +1161,45 @@ class ModuleViewSet(NAVAPIMixin, viewsets.ReadOnlyModelViewSet):
         'device__serial',
     )
     serializer_class = serializers.ModuleSerializer
+
+
+class JWTRefreshViewSet(APIView):
+    """
+    Accepts a valid refresh token.
+    Returns a new refresh token and an access token.
+    """
+
+    def post(self, request):
+        incoming_token = request.data.get('refresh_token')
+        token_hash = hash_token(incoming_token)
+        try:
+            # If hash exists in the database, then we know it is a real token
+            db_token = JWTRefreshToken.objects.get(hash=token_hash)
+        except JWTRefreshToken.DoesNotExist:
+            return Response("Invalid token", status=status.HTTP_403_FORBIDDEN)
```

Comment:
Well, I'm sort of thinking that as long as we can verify that this token was signed by *our* key, it should be treated as valid (until its claims or dates are determined to be invalid, that is).

At the moment, I've entirely forgotten why we need to keep a reference to issued refresh tokens in the database, I just remember pointing out that storing the hash should be enough.  

Only thing I can think of at the moment is that we need it to keep an audit trail?

Reply:
Added comment explicitly stating that looking up the hash in db means we dont have to verify signature